test-suite/src/test/scala/org/scalajs/testsuite/jsinterop/TupleTest.scala

```diff
@@ -0,0 +1,42 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.jsinterop
+
+import org.scalajs.jasminetest.JasmineTest
+
+import scala.scalajs.js
+
+object TupleTest extends JasmineTest {
+
+  describe("scala.scalajs.js.Tuple") {
+
+    it("should provide an equivalent of scala tuple") {
+      val obj: js.Tuple2[Int, String] = (42, "foobar")
```

Comment:
This test is actually testing "should convert from Scala tuple".

There should also be a test directly using `apply`. Also, there should be a test defining the tuple as a `js.Array(stuff...).asInstanceOf[js.Tuple2[Int, String]`, as we want to test that the structure of a tuple is equivalent to a JS array.